### PR TITLE
Updating headline plots and also moving COMINobs to dcom prod dir

### PR DESCRIPTION
**Pull request by PerryShafran-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>

- Removes headline plot driver and J-job scripts and rolls them into the regular plot scripts.  Now only a single J-job in rtofs.
- Places headline plot tar file into its own headline directory under rtofs
- Changes the location of validation directory from dev dcom to prod dcom.  As of this writing, ghrsst and aviso are not yet in production dcom but are planned to be in prod in the next day or two.

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Run the prep job.  Then after prep has been checked, be sure to bring in prep files from emc.vpppg.  Not recommended to merely point to emc.vpppg, you must copy prep from emc.vpppg to your work area.  

Then run all the stats jobs.  ghrsst and aviso may fail if their validation directories haven't moved yet.

Then bring in the emc.vpppg stats directory and run the plots job.  Here you *can* merely point to emc.vpppg in the driver script.  

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

Yes

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

ASAP

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by ShelleyMelchior-NOAA** [comment]
Great instructions. Thanks @PerryShafran-NOAA . I have cloned your fork and checked out branch `feature/rtofs_cleanup`. I will commence testing shortly.

**Comment by PerryShafran-NOAA** [comment]
Please hold off on this for a bit.  I need to make some other fixes plus also deal with the NDBC lookup table issue that was discussed in the email thread.  I will let you know when we can continue.

Perry

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA Adding some comments here of things that can be fixed in this RTOFS PR:

Remove verbosity on lines 48–49: [parm/metplus_config/rtofs/grid2obs/stats/StatAnalysis_fcstRTOFS.conf](https://github.com/NOAA-EMC/EVS/blob/6ba22d2551555704da0de4397128343d260befc3/parm/metplus_config/rtofs/grid2obs/stats/StatAnalysis_fcstRTOFS.conf#L48-L49)
Remove https path and verbosity on lines 49–55:
[parm/metplus_config/rtofs/grid2obs/prep/ASCII2NC_obsARGO.conf](https://github.com/NOAA-EMC/EVS/blob/6ba22d2551555704da0de4397128343d260befc3/parm/metplus_config/rtofs/grid2obs/prep/ASCII2NC_obsARGO.conf#L49-L55)

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA Awesome job here. I've crossed off the J-job fixes in the [EE2 Edits document](https://docs.google.com/document/d/1fYsVrcrjj_NxJBYG6v0sDhi-vYxShT2l2EUauHIByLs/edit?pli=1). The only other EE2 comment from Lin Gan (if you want to do it all in this PR) is to 1) remove the following in the /dev/drivers and 2) make sure that they are defined in the corresponding J-job instead:

export DCOMROOT
export USHevs
export CONFIGevs
export COMINobs
export COMINclimo
export ARCHevs

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA In Lin Gan's first EE2 comment, he said that FIXevs should be removed from all driver scripts too, since it's in the J-jobs. I didn't include it in this list because I thought that removal got included with the last batch of fixes that removed the second instance of FIXevs in the J-jobs, but [it looks like FIXevs is still in the /dev/drivers currently](https://github.com/NOAA-EMC/EVS/commit/3e576db805a8eb008984a1148e834eb7807c3714#diff-c4c85e2d449dfdf9449b7772ccc735865c2d206f92c69b05647a8a0f522cd9dbL39). Sorry for not including it in the latest list too! After FIXevs is removed from the /dev/drivers as well, Shelley can test this PR! Thanks for taking care of this for Samira and for us!

**Comment by PerryShafran-NOAA** [comment]
Oh!  You're right, I never removed that.   Since $HOMEevs/fix is currently a symbolic link to the current location of the fix directory, I think this should work.  I'll remove that in the driver scripts.

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA Thanks! I was able to cross off all the RTOFS lines in the [EE2 compliance checklist](https://docs.google.com/document/d/1fYsVrcrjj_NxJBYG6v0sDhi-vYxShT2l2EUauHIByLs/edit?pli=1)!

**Comment by ShelleyMelchior-NOAA** [comment]
> Oh! You're right, I never removed that. Since $HOMEevs/fix is currently a symbolic link to the current location of the fix directory, I think this should work. I'll remove that in the driver scripts.

Yes! We will put that to test in this PR and then have all the other component drivers updated to remove FIXevs and work from a symlink instead.

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA Including the latest comments about removing "www" lines from the EE2 checklist document:

- Remove line 19 that refers to www, even if commented out: [parm/metplus_config/rtofs/grid2obs/stats/StatAnalysis_fcstRTOFS.conf](https://github.com/NOAA-EMC/EVS/blob/077ab1f895d77091b60636026d7b3b6c5e81cade/parm/metplus_config/rtofs/grid2obs/stats/StatAnalysis_fcstRTOFS.conf#L19)
- Remove line 19 that refers to www, even if commented out:
- [parm/metplus_config/rtofs/grid2grid/stats/GridStat_fcstRTOFS_obsOSISAF_nh.conf](https://github.com/NOAA-EMC/EVS/blob/077ab1f895d77091b60636026d7b3b6c5e81cade/parm/metplus_config/rtofs/grid2grid/stats/GridStat_fcstRTOFS_obsOSISAF_nh.conf#L19)
- Remove line 19 that refers to www, even if commented out:
- [parm/metplus_config/rtofs/grid2grid/stats/GridStat_fcstRTOFS_obsSMAP_climoWOA23.conf](https://github.com/NOAA-EMC/EVS/blob/077ab1f895d77091b60636026d7b3b6c5e81cade/parm/metplus_config/rtofs/grid2grid/stats/GridStat_fcstRTOFS_obsSMAP_climoWOA23.conf#L19)
- Remove line 19 that refers to www, even if commented out: [parm/metplus_config/rtofs/grid2grid/stats/GridStat_fcstRTOFS_obsAVISO_climoHYCOM.conf](https://github.com/NOAA-EMC/EVS/blob/077ab1f895d77091b60636026d7b3b6c5e81cade/parm/metplus_config/rtofs/grid2grid/stats/GridStat_fcstRTOFS_obsAVISO_climoHYCOM.conf#L19)
- Remove line 19 that refers to www, even if commented out: [parm/metplus_config/rtofs/grid2obs/stats/StatAnalysis_fcstRTOFS_obsNDBC.conf](https://github.com/NOAA-EMC/EVS/blob/ef6469e70b418984083d97b110318706444be11f/parm/metplus_config/rtofs/grid2obs/stats/StatAnalysis_fcstRTOFS_obsNDBC.conf#L19)
- Remove line 19 that refers to www, even if commented out: [parm/metplus_config/rtofs/grid2grid/stats/GridStat_fcstRTOFS_obsOSISAF_sh.conf](https://github.com/NOAA-EMC/EVS/blob/ef6469e70b418984083d97b110318706444be11f/parm/metplus_config/rtofs/grid2grid/stats/GridStat_fcstRTOFS_obsOSISAF_sh.conf#L19)
- ​​Remove line 19 that refers to www, even if commented out: [parm/metplus_config/rtofs/grid2grid/stats/GridStat_fcstRTOFS_obsGHRSST_climoWOA23.conf](https://github.com/NOAA-EMC/EVS/blob/ef6469e70b418984083d97b110318706444be11f/parm/metplus_config/rtofs/grid2grid/stats/GridStat_fcstRTOFS_obsGHRSST_climoWOA23.conf#L19)
- ​​Remove line 19 that refers to www, even if commented out: [parm/metplus_config/rtofs/grid2grid/stats/GridStat_fcstRTOFS_obsSMOS_climoWOA23.conf](https://github.com/NOAA-EMC/EVS/blob/ef6469e70b418984083d97b110318706444be11f/parm/metplus_config/rtofs/grid2grid/stats/GridStat_fcstRTOFS_obsSMOS_climoWOA23.conf#L19)

**Comment by PerryShafran-NOAA** [comment]
Thanks for that info; I'll go ahead and remove that stuff.

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA Can you make sure this line gets changed from emc.vpppg to back to $USER? Thanks!

[dev/drivers/scripts/rtofs/plots/jevs_rtofs_plots.sh](https://github.com/NOAA-EMC/EVS/pull/226/files#diff-c4c85e2d449dfdf9449b7772ccc735865c2d206f92c69b05647a8a0f522cd9db)](https://github.com/NOAA-EMC/EVS/pull/226/files#diff-c4c85e2d449dfdf9449b7772ccc735865c2d206f92c69b05647a8a0f522cd9dbR34)

File: dev/drivers/scripts/rtofs/prep/jevs_rtofs_prep.sh [hunk 1]
```diff
@@ -32,19 +32,16 @@ source $HOMEevs/modulefiles/${COMPONENT}/${COMPONENT}_${STEP}.sh
 export VDATE=$(date --date="2 days ago" +%Y%m%d)
 
 export COMIN=/lfs/h2/emc/vpppg/noscrub/$USER/$NET/${evs_ver}
-export COMINobs=/lfs/h1/ops/dev/dcom
 export COMINrtofs=/lfs/h1/ops/$envir/com/$COMPONENT/${rtofs_ver}
 export COMOUT=/lfs/h2/emc/vpppg/noscrub/$USER/$NET/${evs_ver}
 export COMOUTprep=/lfs/h2/emc/vpppg/noscrub/$USER/$NET/${evs_ver}/$STEP/$COMPONENT
+#export DATA=/lfs/h2/emc/ptmp/$USER/$NET/${evs_ver}
```
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/rtofs/prep/jevs_rtofs_prep.sh]: Do we need this line at all? I think it can be removed.

**Comment by ShelleyMelchior-NOAA** [comment]
Hi @PerryShafran-NOAA , I did a quick initial code review and things look pretty good! I will begin testing once we are on cactus.

**Comment by ShelleyMelchior-NOAA** [comment]
@PerryShafran-NOAA , please update the emails to replace lichuan.chen w/ samira.ardani

File: dev/drivers/scripts/rtofs/prep/jevs_rtofs_prep.sh (hunk 1 above)
Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/rtofs/prep/jevs_rtofs_prep.sh]: Took care of this hanging issue.

**Comment by ShelleyMelchior-NOAA** [comment]
prep processing is ongoing -- as you know, it is a long job.

I kicked off the stats jobs. All have completed except for ndbc -- as expected, since that is also a long job.

The stats results are incomplete, however. I ran w/ KEEPDATA=YES so we can do some digging.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/rtofs/rtofs.20230912`
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/`
`jevs_rtofs_argo_stats.o81877144`
`jevs_rtofs_aviso_stats.o81877145`
`jevs_rtofs_ghrsst_stats.o81877146`
`jevs_rtofs_osisaf_stats.o81877148`
`jevs_rtofs_smap_stats.o81877149`
`jevs_rtofs_smos_stats.o81877150`

argo and smos didn't produce the final stats files.

**Comment by PerryShafran-NOAA** [comment]
@ShelleyMelchior-NOAA instead of copying to your own archive of prep files, which is static, please point to the emc.vpppg archive.  Each day's run updates older prep directories, so thus you are missing critical files.  I recommend pointing to the emc.vpppg archive.

You can do this by simply doing this in your prep directory:

ln -sf /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/prep/rtofs rtofs

This should be safe as stats do not write anything in prep directories.

**Comment by ShelleyMelchior-NOAA** [comment]
Ok. I am confused. I thought your instructions said to cp the prep from emc.vpppg into my local.

> Then after prep has been checked, be sure to bring in prep files from emc.vpppg. Not recommended to merely point to emc.vpppg, you must copy prep from emc.vpppg to your work area.

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA @ShelleyMelchior-NOAA 

Please also see Gwen's email, which identifies and issue with the date formatting in this RTOFS PR. 
-------

Shelley,

The date (**2030911**) in this missing data email looks strange.  Could you check the date format?

Thanks,
Gwen

On Thu, Sep 14, 2023 at 12:38 PM shelley melchior <[shelley.melchior@nid001496.cactus.wcoss2.ncep.noaa.gov](mailto:shelley.melchior@nid001496.cactus.wcoss2.ncep.noaa.gov)> wrote:
Warning: No GHRSST OSPO data was available for valid date 2030911.
Missing file is /lfs/h1/ops/prod/dcom/**2030911**/validation_data/marine/ghrsst/**2030911**_OSPO_L4_GHRSST.nc.

**Comment by PerryShafran-NOAA** [comment]
@ShelleyMelchior-NOAA That instruction was needed for prep, since you are writing into prep.  But the reason you are seeing missing files for *stats* is that subsequent runs of prep are bringing in data into the prep directories.

You can rsync from the emc.vpppg into your own archive, and that should work equally well.

**Comment by ShelleyMelchior-NOAA** [comment]
OK, re: stats, I am trashing all my dirs and starting over. I wiggled dates around in multiple runs, some w/ typos, some w/ not, and now I don't have much of a baseline to work with. Please abandon all stats investigation until I can start anew.

FWIW, prep has concluded, but  now I have no idea if how I did it is OK. I may trash all of that, too, at this point.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/rtofs`
logfile: `jevs_rtofs_prep.o81864205`

**Comment by PerryShafran-NOAA** [comment]
Unfortunately there is an error in the prep step.  We'll need to run it again with KEEPDATA=YES so I can see the METplus log file and see what the error is.

**Comment by ShelleyMelchior-NOAA** [comment]
rtofs prep is re-running with KEEPDATA=YES. Please monitor $DATA.
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_prep.o81888314`
DATA=`/lfs/h2/emc/stmp/shelley.melchior/evs_test/prod/tmp/jevs_rtofs_prep.81888314.cbqs01`

**Comment by ShelleyMelchior-NOAA** [comment]
stats is re-running. All datasets are complete except for ndbc. But unfortunately COMOUT is empty!
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/rtofs/rtofs.20230911`
I am swimming in disappointment and confused. I see small stats files, but no final stats files.
logfiles: in `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out` and `ls -ltr jevs_rtofs_*stats*` timestamp 18:07 and younger

**Comment by PerryShafran-NOAA** [comment]
OMG, errors!  

Please set KEEPDATA to YES for all the stat runs. Gosh darn it.

**Comment by ShelleyMelchior-NOAA** [comment]
OK, UPDATE.
All prep and stats jobs have been qdel'd.

Re-running prep w/ VDATE=20230911 and KEEPDATA=YES
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_prep.o81893391`
DATA=`/lfs/h2/emc/stmp/shelley.melchior/evs_test/prod/tmp/jevs_rtofs_prep.81893391.cbqs01`

I will not start any stats processing until prep is complete.

**Comment by PerryShafran-NOAA** [comment]
Also run prep for VDATE=20230910.  I think that might do the trick of bringing in the files that are missing.

**Comment by PerryShafran-NOAA** [comment]
Just checked into the rtofs prep test thus far.  It's looking pretty good, with no errors thus far into the run.  

Perry

**Comment by ShelleyMelchior-NOAA** [comment]
Re-ran prep w/ VDATE=20230910 and KEEPDATA=YES
logfile: /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_prep.o81897871
DATA=/lfs/h2/emc/stmp/shelley.melchior/evs_test/prod/tmp/jevs_rtofs_prep.81897871.cbqs01
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/rtofs`

I will start stats next.

**Comment by PerryShafran-NOAA** [comment]
The prep looks good.  I think prep is fully validated now.

**Comment by PerryShafran-NOAA** [comment]
At this time, I see a stat file for aviso, ghrsst, osisaf, and smap, and their filesizes are as expected.  argo and ndbc each take longer and I think is still running.

At this time I don't see smos, but I don't see smos in the emc.vpppg either.  There seems to be a missing file in ops:  /lfs/h1/ops/prod/dcom/20230910/validation_data/marine/smos/SM_D2023253_Map_SATSSS_data_1day.nc.  @AliciaBentley-NOAA , are we aware of a smos data outage at this time?  

Currently I'm looking at smos in prod/dcom, and it looks like the validation data is sporadic.  I see no files between Sep 8 and Sep 11, a tiny file for Sep 12, and what looks to be a full size file for Sep. 13.  

Looking at the directory, it comes and goes.  The only smos files I see are for Sep 5 through 7 and for Sep 13.  Thus the lack of smos stat files should NOT affect the acceptance of this PR.  In fact, the PR is going as expected given these conditions.

**Comment by ShelleyMelchior-NOAA** [comment]
stats is running now. KEEPDATA=YES and VDATE=20230910.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/rtofs/rtofs.20230910`
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_*stats*.o8192059*`
You can find $DATA in each respective logfile.

In COMOUT, smos is missing, but it is also missing in the emc.vpppg for the same VDATE's COMOUT. So that's promising.
The follow data sets check out: aviso, ghrsst, osisaf, smap.
Just waiting on ndbc and argo.

**Comment by PerryShafran-NOAA** [comment]
BTW, I didn't mention this, but another point of validation for the plots.  The MET_NDBC_STATIONS was properly defined as being from our fix directory, and the buoy check advised by Mallory works perfectly as well.  I wanted to check that specifically because that was new.

**Comment by PerryShafran-NOAA** [comment]
As for rtofs stats, the argo temp file is done and looks good.  Awaiting the argo psal file, which comes a half hour later.  ndbc takes forever and that will take another 2.5 hrs to finish.  I don't think I'm staying up for that.

**Comment by ShelleyMelchior-NOAA** [comment]
Does plots need checking out, too? Shall I get started on running that driver? Looks like yes.

**Comment by PerryShafran-NOAA** [comment]
Plots has some changes (mainly EE2 stuff).  And yes, the original purpose of this PR was the headline plots, which has the headline tar file in its own directory, plus it condenses the J-jobs into one (there were two J-jobs before).

**Comment by ShelleyMelchior-NOAA** [comment]
plots processing is underway for VDATE=20230910 and KEEPDATA=YES and using emc.vpppg stats.
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_plots.o81925582`

**Comment by ShelleyMelchior-NOAA** [comment]
plots processing is completed!
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_plots.o81925582`
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/rtofs/rtofs.20230910`
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/rtofs/headline.20230910`
Let me know if you can't expand the tar files.

**Comment by ShelleyMelchior-NOAA** [comment]
heading to bed ... ndbc stats keeps churning. Probably about halfway through. We'll check first thing in the morning!
`81920594.cbqs01                shelley.melchi* dev             jevs_rtofs_ndb*   165145    1     1  500gb 06:00 R 02:52`

**Comment by PerryShafran-NOAA** [comment]
All stats files are there and good.  

Plots:  argo and ndbc are empty!!!  I'm looking into why that is.  All other plots tar files, including headline, are OK.

Here is an issue - I have no idea why there is a missing ush file:

`python: can't open file '/lfs/h2/emc/vpppg/noscrub/shelley.melchior/githubwkspc/forks/PSrtofs/EVS/ush/time_series.py': [Errno 2] No such file or directory`

And I see the reason why.  I need to make a change.

Perry

**Comment by PerryShafran-NOAA** [comment]
Please run plots again.

**Comment by ShelleyMelchior-NOAA** [comment]
Glad all of the stats check out! Will pull in your changes and re-run plots.

**Comment by ShelleyMelchior-NOAA** [comment]
rtofs plots is re-running after pulling in the updates. KEEPDATA=YES and VDATE=20230910 and using emc.vpppg stats.
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_plots.o81963153`

**Comment by PerryShafran-NOAA** [comment]
While we were dealing with all the other PRs, I wanted to note that I am seeing all the plot tarballs for rtofs.  They are the size that are expected too.

The headline part still seems to be running. 

I would like to see one grid2grid plot (say for aviso) and one grid2obs plot (say for argo) to check for logos.

**Comment by PerryShafran-NOAA** [comment]
Oh ok, the plots is still running!  Because I see argo and ndbc plots that exist.  Oops!  Needless alarm.  The final headline stat file will probably be good then (had I seen the time stamp, that would have been a good clue that you're still running).  

I still want to see plots to check for logos.

**Comment by PerryShafran-NOAA** [comment]
OK, the headline plot tarball looks to be of the right size so that's good.

**Comment by ShelleyMelchior-NOAA** [comment]
rtofs plots completed. I can't compare files sizes for plots on 20230910 in the emv.vpppg parallel b/c there are no plots for that day in ptmp on cactus. They are on dogwood. They aren't exact matches, but close enough.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/rtofs/rtofs.20230910`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_plots.o81963153`

I unpacked the argo and aviso tar files. I see logos.
https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/rtofs/evs.rtofs.acc.salinity_z50_argo.last60days.timeseries_valid00z_f000.glb.png
https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/rtofs/evs.rtofs.me.ssh_z0_aviso.last60days.timeseries_valid00z_f168.npac.png

COMOUT for headline = `/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/rtofs/headline.20230910`
Need me to untar that one, or are we good?

**Comment by PerryShafran-NOAA** [comment]
Logos!

I think that we don't need that since the headline scores run on the same configuration files that the other plot files do.  Therefore, I am good to merge this one finally.

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
This code has been tested and expected results were obtained. This PR is approved for merge.

**Comment by ShelleyMelchior-NOAA** [comment]
I did code review as we went along, so the official code review is now complete.

**Comment by ShelleyMelchior-NOAA** [comment]
Just need one sync fork in `feature/rtofs_cleanup`.

**Pull request merged**